### PR TITLE
Update Wage Categories when pushing e/f to MYOB

### PR DESCRIPTION
**Pull request by thukim**
- https://trello.com/c/KM0zZc0K/528-pushing-correct-pay-rate-type-to-myob-when-create-new-e-f-from-eh

### CODE REVIEW

File: lib/eh/myob/data/member_mapper.rb [hunk 1]
```diff
@@ -102,6 +117,17 @@ def myob_wage_info(eh_attrs)
                       })
           end
 
+          def myob_wage_categories(current_wage, hourly_wc)
+            current_wage_categories = current_wage['WageCategories'] || []
+            if current_wage['PayBasis'] == HOURLY && hourly_wc.present? &&
+              current_wage_categories.detect { |wc| wc['Name'] == MYOB_BASE_HOURLY }.blank?
+              current_wage_categories << hourly_wc.slice(*WAGE_FIELDS) if hourly_wc
```
Comment (mmeeoorroo) [inline lib/eh/myob/data/member_mapper.rb]: why don't you merge `if hourly_wc` to the outer `if`
```ruby
if current_wage['PayBasis'] == HOURLY 
   && hourly_wc.present? 
   && current_wage_categories.detect { |wc| wc['Name'] == MYOB_BASE_HOURLY }.blank?
   && hourly_wc
  current_wage_categories << hourly_wc.slice(*WAGE_FIELDS)
end
```

Comment (mmeeoorroo) [reply lib/eh/myob/data/member_mapper.rb]: wait, there is check `hourly_wc.present?` already, so we can strip the last one

File: lib/eh/myob/data/member_mapper.rb [hunk 2]
```diff
@@ -66,7 +78,7 @@ def new_myob_employee(eh_attrs)
             }.merge(myob_convert_status(eh_attrs))
           end
 
-          def myob_payroll_detail_map(eh_attrs, myob_payroll_attrs)
+          def myob_payroll_detail_map(eh_attrs, myob_payroll_attrs, hourly_wc)
```
Comment (mmeeoorroo) [inline lib/eh/myob/data/member_mapper.rb]: I'm confused. The trello card mentions both `Base hourly` and `Base salary`, but I don't see `Base salary` in this PR.

Comment (thukim) [reply lib/eh/myob/data/member_mapper.rb]: `Base salary` is the default one:
`When I create an E/F in EH and pay rate type is "Annum", an employee card is created in MYOB and "Base Salary" is ticked.` as described in the card

File: lib/eh/myob/data/member_mapper.rb (hunk 1 above)
Comment (thukim) [reply lib/eh/myob/data/member_mapper.rb]: yup, the `if` at line 124 is redundant

File: lib/eh/myob/data/member_mapper.rb (hunk 2 above)
Comment (mmeeoorroo) [reply lib/eh/myob/data/member_mapper.rb]: Got it

**Pull request merged**